daq/runner.py

```diff
@@ -926,18 +926,26 @@ def _send_device_result(self, mac, test_results):
         self._device_result_client.send_device_result(mac, device_result)
 
     def _calculate_device_result(self, test_results):
+        failed = False
```

Comment:
I am not sure why we always check all the three items: exception, return code, and module test results, since one failure is already enough to indicate that the test has failed. Is it because we want to have a more verbose logging on the test results?